artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -1413,8 +1500,18 @@ private void installPageTransaction(final Transaction tx, final RouteContextList
       return;
    }
 
+   @Override
+   public boolean hasPendingIO() {
+      return timedWriter != null && timedWriter.hasPendingIO();
```

Comment:
When timedWriter is null, could there be writes _happening presently_ which this doesnt account for?

Reply:
when timedWriter is null, we have the previous semantic. Meaning it needs to return false, as no pending writes and it can clear page status without issues.